src/scipp/core/variable.py

```diff
@@ -726,10 +726,17 @@ def arange(dim: str,
            *,
            unit: Union[Unit, str, None] = default_unit,
            dtype: Optional[DTypeLike] = None) -> Variable:
-    """Constructs a :class:`Variable` with evenly spaced values within a given
-    interval.
-    Values are generated within the half-open interval [start, stop)
-    (in other words, the interval including start but excluding stop).
+    """Creates a :class:`Variable` with evenly spaced values within a given interval.
+
+    Values are generated within the half-open interval [start, stop).
+    In other words, the interval including start but excluding stop.
+
+    ``start``, ``stop``, and ``step`` may be given as plain values or as 0-D variables.
+    In the latter case this then implies the unit.
```

Comment:
The unit can still be overridden by the `unit` arg. But start, stop, step must all have the same unit unless one is specified explicitly.